bundles/org.openhab.binding.playstation/src/main/java/org/openhab/binding/playstation/internal/discovery/PlayStationDiscovery.java

```diff
@@ -0,0 +1,475 @@
+import java.net.UnknownHostException;
+import java.nio.charset.StandardCharsets;
+import java.util.HashMap;
+import java.util.Map;
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.eclipse.jdt.annotation.Nullable;
+import org.openhab.core.config.discovery.AbstractDiscoveryService;
+import org.openhab.core.config.discovery.DiscoveryResult;
+import org.openhab.core.config.discovery.DiscoveryResultBuilder;
+import org.openhab.core.config.discovery.DiscoveryService;
+import org.openhab.core.net.NetworkAddressService;
+import org.openhab.core.thing.Thing;
+import org.openhab.core.thing.ThingUID;
+import org.openhab.core.util.HexUtils;
+import org.osgi.service.component.annotations.Component;
+import org.osgi.service.component.annotations.Reference;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * The {@link PlayStationDiscovery} is responsible for discovering
+ * all PS4 devices
+ *
+ * @author Fredrik Ahlström - Initial contribution
+ */
+@NonNullByDefault
+@Component(service = { DiscoveryService.class, PlayStationDiscovery.class }, configurationPid = "binding.playstation")
+public class PlayStationDiscovery extends AbstractDiscoveryService {
+
+    private final Logger logger = LoggerFactory.getLogger(PlayStationDiscovery.class);
+
+    private static final int DISCOVERY_TIMEOUT_SECONDS = 2;
+
+    private @Nullable NetworkAddressService networkAS;
+
+    public PlayStationDiscovery() {
+        super(SUPPORTED_THING_TYPES_UIDS, DISCOVERY_TIMEOUT_SECONDS * 2, true);
+    }
+
+    @Override
+    protected void startScan() {
+        logger.debug("Updating discovered things (new scan)");
+        discover();
+        discoverPS3();
+    }
+
+    @Reference
+    public void bindNetworkAddressService(NetworkAddressService network) {
+        networkAS = network;
+    }
+
+    private @Nullable InetAddress getBroadcastAdress() {
+        NetworkAddressService nwService = networkAS;
+        if (nwService != null) {
+            try {
+                String address = nwService.getConfiguredBroadcastAddress();
+                if (address != null) {
+                    return InetAddress.getByName(address);
+                } else {
+                    return InetAddress.getByName("255.255.255.255");
+                }
+            } catch (UnknownHostException e) {
+                // We catch errors later.
+            }
+        }
+        return null;
+    }
+
+    private @Nullable InetAddress getIPv4Adress() {
+        NetworkAddressService nwService = networkAS;
+        if (nwService != null) {
+            try {
+                String address = nwService.getPrimaryIpv4HostAddress();
+                if (address != null) {
+                    return InetAddress.getByName(address);
+                }
+            } catch (UnknownHostException e) {
+                // We catch errors later.
+            }
+        }
+        return null;
+    }
+
+    private synchronized void discover() {
```

Comment:
```suggestion
    private synchronized void discoverPS4() {
```

Reply:
Hopefully this won't be needed, we'll see November 19.

Reply:
Fixed.